base_layer/core/src/validation/aggregate_body/aggregate_body_chain_validator.rs

```diff
@@ -110,27 +110,44 @@ fn validate_input_not_pruned<B: BlockchainBackend>(
     db: &B,
 ) -> Result<Vec<TransactionInput>, ValidationError> {
     let mut inputs: Vec<TransactionInput> = body.inputs().clone();
+    let outputs: Vec<TransactionOutput> = body.outputs().clone();
     for input in &mut inputs {
         if input.is_compact() {
-            let output_mined_info = db
-                .fetch_output(&input.output_hash())?
-                .ok_or(ValidationError::UnknownInput)?;
+            let output = match db.fetch_output(&input.output_hash()) {
+                Ok(val) => match val {
+                    Some(output_mined_info) => output_mined_info.output,
+                    None => {
+                        let input_output_hash = input.output_hash();
+                        if let Some(found) = outputs.iter().find(|o| o.hash() == input_output_hash) {
+                            found.clone()
+                        } else {
```

Comment:
nit(reduce-memory-usage): I think you can avoid allocating a new vector for all outputs

```suggestion
                        let input_output_hash = input.output_hash();
                        if let Some(found) = body.outputs().iter().find(|o| o.hash() == input_output_hash) {
                            found.clone()
                        } else {
```

And you can remove the clone above.
`    let outputs: Vec<TransactionOutput> = body.outputs().clone();`